src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +882,231 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
     TestSigma1Parsing(mem, bufferSize, Sigma1SessionIdTooBig);
 }
 
+TEST_F(TestCASESession, EncodeSigma1Test)
+{
+    CASESessionAccess::EncodeSigma1Inputs encodeParams;
+
+    uint8_t random[32];
+    EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&random[0], sizeof(random)), CHIP_NO_ERROR);
+    encodeParams.initiatorRandom    = ByteSpan(random, sizeof(random));
+    encodeParams.initiatorSessionId = 7315;
+    uint8_t destinationId[32]       = { 0xDE, 0xAD };
+    encodeParams.destinationId      = ByteSpan(destinationId, sizeof(destinationId));
+
+    ReliableMessageProtocolConfig mrpConfig = GetDefaultMRPConfig();
+    encodeParams.initiatorMrpConfig         = &mrpConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 should fail when there is no public key
+        EXPECT_EQ(CHIP_ERROR_INVALID_ARGUMENT, CASESessionAccess::EncodeSigma1(msg, encodeParams));
+    }
+
+    Crypto::P256Keypair * ephemeralKey = gDeviceOperationalKeystore.AllocateEphemeralKeypairForCASE();
+    ASSERT_NE(ephemeralKey, nullptr);
+    EXPECT_EQ(CHIP_NO_ERROR, ephemeralKey->Initialize(ECPKeyTarget::ECDH));
+    encodeParams.initiatorEphPubKey = &ephemeralKey->Pubkey();
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 will Succeed when Public Key is provided
+        EXPECT_EQ(CHIP_NO_ERROR, CASESessionAccess::EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 should fail when MRP config is missing
+        encodeParams.initiatorMrpConfig = nullptr;
+        EXPECT_EQ(CHIP_ERROR_INVALID_ARGUMENT, CASESessionAccess::EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // Succeed when MRP Config is provided
+        encodeParams.initiatorMrpConfig = &mrpConfig;
+        EXPECT_EQ(CHIP_NO_ERROR, CASESessionAccess::EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg1;
+
+        // Round Trip Test: Encode Sigma1, Parse it then verify parsed values
+        EXPECT_EQ(CHIP_NO_ERROR, CASESessionAccess::EncodeSigma1(msg1, encodeParams));
+
+        System::PacketBufferTLVReader tlvReader;
+        tlvReader.Init(std::move(msg1));
+
+        CASESessionAccess session;
+        CASESessionAccess::ParsedSigma1 parseParams;
```

Comment:
```suggestion
        CASESessionAccess::ParsedSigma1 parsedMessage;
```
This was https://github.com/project-chip/connectedhomeip/pull/36679#discussion_r1889548097

Reply:
don't know what happened, its fixed now